corporate/lib/stripe.py

```diff
@@ -2069,14 +2069,31 @@ def get_initial_upgrade_context(
 
         return None, context
 
+    def min_licenses_for_flat_discount_to_self_hosted_basic_plan(
+        self, customer: Optional[Customer]
+    ) -> int:
+        # Since monthly and annual TIER_SELF_HOSTED_BASIC plans have same per user price we only need to do this calculation once.
+        # If we decided to apply this for other tiers, then we will have to do this calculation based on billing schedule selected by the user.
+        price_per_license = get_price_per_license(
+            CustomerPlan.TIER_SELF_HOSTED_BASIC, CustomerPlan.BILLING_SCHEDULE_MONTHLY
+        )
+        if customer is None:
+            return (
+                Customer._meta.get_field("flat_discount").get_default() // price_per_license
+            ) + 1
```

Comment:
How does this line work? It seems like it returns a discount, without the division in the `elif` clause...

Reply:
I am not sure what you mean. The first check is for a new customer where we don't have a customer entry, so we take the default discount and divide the by the price per license of the tier. In the `elif` clause, where customer exists, we just use the `flat_discount` that was set for the customer to calculate the number of minimum licenses.

Reply:
Oh, I somehow misread the `// price_per_license` as a JS comment. OK.